### PR TITLE
Record: 11L EMA + Int6 + XSA + LeakyReLU² + Partial RoPE (val_bpb: 1.1309)

### PR DESCRIPTION
**Pull request by parinzee**
## Summary
- **val_bpb: 1.1309** (mean of 3 seeds, std: 0.00017)
- 11 layers, 512 dim, 8H/4KV GQA
- Artifact: ~15.8 MB (all seeds under 16 MB)

## 3-Seed Results

| Seed | val_bpb | artifact_bytes |
|------|---------|----------------|
| 42   | 1.13109 | 15,764,564     |
| 1337 | 1.13085 | 15,626,741     |
| 2024 | 1.13067 | 15,923,256     |
| **Mean** | **1.13087** | |
| Std  | 0.00017 | |

## Key Changes from Baseline

1. **11 layers** (up from 10), 512 dim, 8 heads / 4 KV heads (GQA)
2. **XSA (Exclusive Self Attention)** on last 4 layers for better representation
3. **LeakyReLU(0.5)²** activation — squared leaky ReLU with 0.5 negative slope
4. **Partial RoPE** — only 16/64 dims use rotary embeddings
5. **EMA weight averaging** (decay=0.997) for smoother final weights
6. **Int6 quantization** for all large weight matrices + zstd-22 compression
7. **Scale clamping fix** — clamp_min(1/clip_range) improves quantization quality
8. **Smaller batch size** (524288 tokens) to fit more training steps (~8200 steps in 600s)
9. **BigramHash(2048, dim=128)** token embeddings
10. **warmdown_iters=4500** for learning rate schedule
11. **Higher learning rates** (matrix_lr=0.025, scalar_lr=0.025)

## Run Command

```bash
torchrun --standalone --nproc_per_node=8 train_gpt.py
```

Built on SOTA baseline by @thwu1 (PR #180).

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
### Community Review — Record: 11L EMA + Int6 + XSA + LeakyReLU² + Partial RoPE (val_bpb: 1.1309)

**BPB:** 1.1309 | **Compliance:** LOOKS CLEAN — pure-neural submission, no TTT/SLOT/n-gram-cache

**What I found in the code** (head SHA `739a729f6b4a`, file `records/track_10min_16mb/2026-03-23_11L_EMA_Int6_XSA_LeakyReLU_PartialRoPE/train_gpt.py`):

Static code review found no TTT adaptation function, no SLOT optimization loop, no n-gram-cache class, and no pre-quant val-token fine-tune. The eval path uses the standard sliding-window stride-64 pattern. The submission is a pure-neural architecture iteration on the standard SP1024/SP4096/SP8192 baseline.

**CPU smoke test (CT2038 proteus-engine, 2026-04-11):** import OK in 0.03s, dim=512, layers=10, vocab=1024, code=53603 B, SMOKE_TEST_PASS

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending the usual record-track checks (3-seed validation, under-16MB artifact cap, ≤600s train + ≤600s eval on 8×H100 SXM). No compliance flags from the classification pass — this looks like a clean pure-neural iteration on the standard baseline.

*Auto-classification caveat:* this review was drafted by the AST-based classifier. If there's a non-standard eval mechanism (logit postprocessing, hedge mixing, etc.) that I missed because it's factored into a helper file or a non-standard function name, please flag it and I'll re-run the audit manually.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). CPU smoke test (CT2038 proteus-engine, 2026-04-11): import OK in 0.03s, dim=512, layers=10, vocab=1024, code=53603 B, SMOKE_TEST_PASS. Classification via deterministic AST-based `classify_prs.py` (pattern bank derived from ~65 manually-reviewed PRs earlier in the 2026-04-11 sweep). This review was auto-drafted from a template and spot-checked before posting — if the template misread your code, please call it out so I can iterate the classifier.*
